.github/workflows/release_docker.yml

```diff
@@ -0,0 +1,114 @@
+          echo "::set-output name=patch::$PATCH"
+
+      - name: Is pkg 'latest'
+        id: latest
+        uses: ./.github/actions/latest-wrangler
+        with:
+          package: ${{ github.event.inputs.package }}
+          new_version: ${{ github.event.inputs.version_number }}
+          gh_token: ${{ secrets.GITHUB_TOKEN }}
+          halt_on_missing: False
+
+  setup_image_builder:
+    name: Set up docker image builder
+    runs-on: ubuntu-latest
+    needs: [get_version_meta]
+    steps:      
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+  build_and_push:
+    name: Build images and push to GHCR
+    runs-on: ubuntu-latest
+    needs: [setup_image_builder, get_version_meta]
+    steps:
+      - name: Get docker build arg
+        id: build_arg
+        run: |
+          echo "::set-output name=build_arg_name::"$(echo ${{ github.event.inputs.package }} | sed 's/\-/_/g')
+          echo "::set-output name=build_arg_value::"$(echo ${{ github.event.inputs.package }} | sed 's/postgres/core/g')
+
+      - name: Log in to the GHCR
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push PATCH
+        uses: docker/build-push-action@v2
+        with:
+          file: docker/Dockerfile
+          push: True
+          target: ${{ github.event.inputs.package }}
+          build-args: |
+            ${{ steps.build_arg.outputs.build_arg_name }}_ref=${{ steps.build_arg.outputs.build_arg_value }}@v${{ github.event.inputs.version_number }}
+          tags: |
+            ghcr.io/dbt-labs/${{ github.event.inputs.package }}:${{ github.event.inputs.version_number }}
+
+      - name: Build and push MINOR.latest
+        uses: docker/build-push-action@v2
+        if: ${{ needs.get_version_meta.outputs.minor_latest == 'True' }}
+        with:
+          file: docker/Dockerfile
+          push: True
+          target: ${{ github.event.inputs.package }}
+          build-args: |
+            ${{ steps.build_arg.outputs.build_arg_name }}_ref=${{ steps.build_arg.outputs.build_arg_value }}@v${{ github.event.inputs.version_number }}
+          tags: |
+            ghcr.io/dbt-labs/${{ github.event.inputs.package }}:${{ needs.get_version_meta.outputs.major }}.${{ needs.get_version_meta.outputs.minor }}.latest
+
+      - name: Build and push latest
+        uses: docker/build-push-action@v2
+        if: ${{ needs.get_version_meta.outputs.latest == 'True' }}
+        with:
+          file: docker/Dockerfile
+          push: True
+          target: ${{ github.event.inputs.package }}
+          build-args: |
+            ${{ steps.build_arg.outputs.build_arg_name }}_ref=${{ steps.build_arg.outputs.build_arg_value }}@v${{ github.event.inputs.version_number }}
+          tags: |
+            ghcr.io/dbt-labs/${{ github.event.inputs.package }}:latest
```

Comment:
These last 3 steps all build the same image but then push it with a different tag, correct? Does the `tags` field take multiple tags? Can we build and push once and figure out the tags to use in a previous step?

Reply:
It does and we could, but docker is smart enough not to push or build anything that it already has at the given location so the extraneous build/pushes are essentially no-ops as long as they are contained in the same job.

Reply:
Ohhhh ok good to know. I thought we were building again every time and that is what bothered me so 👍 with keeping it this way